commandapi-platforms/commandapi-bukkit/commandapi-bukkit-plugin-mojang-mapped/src/main/java/dev/jorel/commandapi/CommandAPIMain.java

```diff
@@ -133,4 +138,51 @@ private JavaPlugin getAndValidatePlugin(String pluginName) {
 	public void onEnable() {
 		CommandAPI.onEnable();
 	}
+
+	/**
+	 * In contrast to the superclasses method {@link org.bukkit.plugin.java.JavaPlugin#saveDefaultConfig()},
+	 * this doesn't fail silently if the config.yml already exists but instead will update the config with
+	 * new values if available.
+	 * <p>
+	 * This should fail silently if all values are set already.
+	 */
+	@Override
+	public void saveDefaultConfig() {
```

Comment:
It looks like the logic in this method for checking if the config file exists and either creating or updating is shared with CommandAPIVelocity (and with the spigot-mapped plugin). I think this could be merged in `ConfigGenerator` using a method in `ConfigurationAdapter` that loads its platform-specific configuration object from a file.